packages/flutter_tools/test/web.shard/test_data/hot_restart_web_test_common.dart

```diff
@@ -119,6 +97,16 @@ Future<void> _testProject(
             '--no-web-experimental-hot-reload',
         ],
       );
+      // hot restart works without error
+      await flutter.hotRestart();
+
+      final Completer<void> completer = Completer<void>();
```

Comment:
This still needs to go before we do a hot restart I believe. It is possible that the hot restart happens fast enough that we miss the stdout result because we didn't subscribe to it early enough.

Reply:
Ah okay, then the move to listen before the second hot restart is still sufficient. Thanks for looking into it!